jdbc-source/src/test/java/org/springframework/cloud/stream/module/jdbc/source/JdbcSourceIntegrationTests.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.jdbc.source;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jdbc.EmbeddedDataSourceConfiguration;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Integration Tests for JdbcSource. Uses hsqldb as a (real) embedded DB.
+ *
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {JdbcSourceApplication.class, EmbeddedDataSourceConfiguration.class})
+@IntegrationTest({"server.port=-1"})
```

Comment:
OK! Without `@WebIntegrationTest` we end up like:
```
DispatcherServletAutoConfiguration did not match
- @ConditionalOnClass classes found: org.springframework.web.servlet.DispatcherServlet (OnClassCondition)
- not a web application (OnWebApplicationCondition)
```
But the same result we get even without this noisy top-level `@IntegrationTest`.
In this scenario it does not have any effect: independently of that `server.port` value we don't step into the `ServerProperties` at all.

What is funny here that with the `@WebIntegrationTest({"server.port=-1"})` we end up with the:
```
2016-01-19 13:15:38.946  INFO 8796 --- [           main] s.b.c.e.t.TomcatEmbeddedServletContainer : Tomcat started on port(s): -1 (http)
```

Although the Boot Docs say:
> To switch off the HTTP endpoints completely, but still create a WebApplicationContext, use server.port=-1 (this is sometimes useful for testing).

However we have the protection code in the `TomcatEmbeddedServletContainerFactory`:
```
protected void customizeConnector(Connector connector) {
		int port = (getPort() >= 0 ? getPort() : 0);
		connector.setPort(port);
```

Maybe that "endpoints disabling" doesn't relate to the Embedded Servlet Container...
Although yes: it isn't started:
```
return new TomcatEmbeddedServletContainer(tomcat, getPort() >= 0);
....
public void start() throws EmbeddedServletContainerException {
	try {
		addPreviouslyRemovedConnectors();
		Connector connector = this.tomcat.getConnector();
		if (connector != null && this.autoStart) {
			startConnector(connector);
              }
```

So, I guess we can remove this annotation from here, but for other tests when we would like to share some options for all tests, like we do in the `CassandraSinkIntegrationTests`, we don't have choice unless use `@WebIntegrationTest`. Otherwise the downstream `@IntegrationTest` overrides any properties from the upstream.

But I agree that it would be good to find some compromise between real world testing and the entire Spring Boot noise in the particular test.